src/infrastructure/database/migrations/20250324184819_add_status_to_tasks.ts

```diff
@@ -0,0 +1,21 @@
+import type { Knex } from "knex";
+
+enum TaskStatus {
+    DONE = 'DONE',
+    LATE = 'LATE'
+}
+
+export async function up(knex: Knex): Promise<void> {
+    await knex.schema.alterTable('tasks', (table) => {
+        table.enum('status', Object.values(TaskStatus))
+          .nullable()
+      });
+}
```

Comment:
Entendi o que cê fez aqui, caso uma task não tenha status (status == null) ela é considerada como open, certo?

Não há nada de estranho nessa abordagem, mas Enums ficam mais redondinhos não sendo nullable e tendo um status padrão como default. Isso garante que toda tarefa sempre terá um status explícito, evitando a necessidade de interpretar NULL como um estado específico

```suggestion
enum TaskStatus {
    OPEN = 'OPEN',
    DONE = 'DONE',
    LATE = 'LATE'
}

export async function up(knex: Knex): Promise<void> {
    await knex.schema.alterTable('tasks', (table) => {
        table.enum('status', Object.values(TaskStatus))
          .nullable()
          .defaultTo(TaskStatus.OPEN)
      });
}
```

caso cê tenha outros registros no BD, pode exigir um pouco mais de trabalho pra migrar os dados para a nova coluna.